diff --git a/app-office/libreoffice/libreoffice-26.8.0.3.ebuild b/app-office/libreoffice/libreoffice-26.8.0.3.ebuild
index 7525f93..9fa4384 100644
--- a/app-office/libreoffice/libreoffice-26.8.0.3.ebuild
+++ b/app-office/libreoffice/libreoffice-26.8.0.3.ebuild
@@ -87,7 +87,8 @@ KEYWORDS="~amd64 ~arm ~arm64 ~loong ~ppc64 ~riscv ~x86"
 LO_EXTS="nlpsolver scripting-beanshell scripting-javascript wiki-publisher"
 
 IUSE="accessibility base bluetooth +branding coinmp +cups custom-cflags +dbus debug eds
-googledrive gstreamer gtk3 +gtk4 kde ldap +mariadb odk pdfimport postgres qt6 test valgrind vulkan
+googledrive gstreamer gtk3 +gtk4 kde ldap +mariadb odk pdfimport postgres qt6 test valgrind
+vulkan wayland
 $(printf 'libreoffice_extensions_%s ' ${LO_EXTS})"
 
 REQUIRED_USE="${PYTHON_REQUIRED_USE}
@@ -98,6 +99,7 @@ REQUIRED_USE="${PYTHON_REQUIRED_USE}
 	libreoffice_extensions_scripting-beanshell? ( java )
 	libreoffice_extensions_scripting-javascript? ( java )
 	libreoffice_extensions_wiki-publisher? ( java )
+	wayland? ( || ( gtk3 gtk4 ) )
 "
 
 RESTRICT="!test? ( test )"
@@ -196,7 +198,8 @@ COMMON_DEPEND="${PYTHON_DEPS}
 		dev-libs/glib:2
 		gnome-base/dconf
 		media-libs/mesa[egl(+)]
-		x11-libs/gtk+:3[wayland,X]
+		wayland? ( x11-libs/gtk+:3[X,wayland] )
+		!wayland? ( x11-libs/gtk+:3[X] )
 		x11-libs/pango
 	)
 	gtk4? (
@@ -204,7 +207,8 @@ COMMON_DEPEND="${PYTHON_DEPS}
 			dev-libs/glib:2
 			gnome-base/dconf
 			media-libs/mesa[egl(+)]
-			gui-libs/gtk:4[wayland,X]
+			wayland? ( gui-libs/gtk:4[X,wayland] )
+			!wayland? ( gui-libs/gtk:4[X] )
 			x11-libs/pango
 	)
 	kde? (
@@ -607,6 +611,10 @@ src_configure() {
 
 	tc-is-lto && myeconfargs+=( --enable-lto )
 
+	# defang automagic dependencies
+	#use X || append-flags -DGENTOO_GTK_HIDE_X11
+	use wayland || append-flags -DGENTOO_GTK_HIDE_WAYLAND
+
 	MARIADBCONFIG="$(type -p $(usex mariadb mariadb mysql)_config)" \
 	econf "${myeconfargs[@]}"
 }
